In [1]:
!pip install bs4
!pip install requests

In [1]:
from bs4 import BeautifulSoup
import requests
import pandas as pd
import re

1) Write a python program to display all the header tags from wikipedia.org.

In [3]:
def display_header_tags(url):
    page=requests.get(url)
    if page.status_code==200:
        soup=BeautifulSoup(page.content,'html.parser')
        header_tags=soup.find_all(re.compile(r'^h[1-6]$'))
        for tag in header_tags:
            print(tag.text.strip())
    else:
        print("Can't fetch the data.")
url='https://en.wikipedia.org/wiki/Main_Page' #Wikipedia's main page
display_header_tags(url)

Main Page
Welcome to Wikipedia
From today's featured article
Did you know ...
In the news
On this day
Today's featured picture
Other areas of Wikipedia
Wikipedia's sister projects
Wikipedia languages


2) Write a python program to display IMDB’s Top rated 100 movies’ data (i.e. name, rating, year of release) and make data frame.

In [4]:
page=requests.get("https://www.imdb.com/chart/top")

In [5]:
page

<Response [403]>

We are getting the response status code for this website as 403. This indicates that scraping the data from this website is not possible.

3) Write a python program to scrape mentioned details from dineout.co.in : i) Restaurant name ii) Cuisine iii) Location iv) Ratings v) Image URL.

In [2]:
def scrape_restaurant_details(url):
    page=requests.get(url)
    if page.status_code==200:
        soup=BeautifulSoup(page.content)
        restaurant_containers = soup.find_all('div', class_='restnt-card restaurant')
        restaurants = []
        for restaurant in restaurant_containers:
            name = restaurant.find('a').text.strip()
            cuisines_text = restaurant.find('span', class_='double-line-ellipsis').text.strip()
            cuisines = re.sub(r'^.*?\|', '', cuisines_text).strip()  # Using regex to remove the unwanted part
            location = restaurant.find('div', class_ = 'restnt-loc ellipsis').text.strip()
            rating=restaurant.find('div', class_=lambda x: x and x.startswith('restnt-rating')).text.strip()
            image_url=restaurant.find('img',class_='no-img')['data-src']
            restaurants.append((name,cuisines,location,rating,image_url))
        return restaurants
    else:
        print("Can't fetch the data.")
        
url="https://www.dineout.co.in/delhi-restaurants"
restaurants=scrape_restaurant_details(url)            
for name, cuisines,location,rating,image_url in restaurants:
    print("Restaurant:",name)
    print("Cuisines:",cuisines)
    print("Location:",location)
    print("Rating:",rating)
    print("Image URL:",image_url)
    print()

Restaurant: Delia My Bar Headquarters
Cuisines: North Indian, Chinese, Fast Food, Beverages
Location: Karol Bagh, Central Delhi
Rating: 4.1
Image URL: https://im1.dineout.co.in/images/uploads/restaurant/sharpen/1/d/m/p133271-170455034765995fcb87ba1.jpg?tr=tr:n-medium

Restaurant: The G.T. Road
Cuisines: North Indian
Location: M-Block,Connaught Place, Central Delhi
Rating: 4.3
Image URL: https://im1.dineout.co.in/images/uploads/restaurant/sharpen/3/l/l/p31393-171083403865f94176e32c0.jpg?tr=tr:n-medium

Restaurant: My Bar Headquarters
Cuisines: North Indian, Chinese
Location: Connaught Place, Central Delhi
Rating: 4
Image URL: https://im1.dineout.co.in/images/uploads/restaurant/sharpen/1/d/f/p19636-16463919576221f295ae1f1.jpg?tr=tr:n-medium

Restaurant: Connaught Club House
Cuisines: North Indian, Asian, Continental, Mexican
Location: Janpath, Central Delhi
Rating: 4.4
Image URL: https://im1.dineout.co.in/images/uploads/restaurant/sharpen/9/y/f/p97486-170556387265a8d6e0e2ea7.jpg?tr=tr:n-

4) Write s python program to display list of respected former finance minister of India(i.e. Name , Term of office) from https://presidentofindia.nic.in/former-presidents.htm and make data frame.

The website is https://presidentofindia.nic.in/former-presidents and only the names and numbers of former presidents of India can be scraped from it. Hence, writing a Python program to display a list of the names and numbers of former presidents of India and making a data frame.

In [3]:
def display_former_presidents_list(url):
    page=requests.get(url)
    if page.status_code==200:
        soup=BeautifulSoup(page.content)
        names=[]
        numbers=[]
        for i in soup.find_all('h3'):
            name=i.text.strip()
            names.append(name)
        for i in soup.find_all('h5'):
            number=i.text.strip()
            numbers.append(number)
        former_presidents={'Name':names,'Number':numbers}
        former_presidents_df=pd.DataFrame(former_presidents)
        print(former_presidents_df)
    else:
        print("Can't fetch the data.")
    
url="https://presidentofindia.nic.in/former-presidents"
display_former_presidents_list(url)

                            Name                   Number
0           Shri Ram Nath Kovind  14th President of India
1          Shri Pranab Mukherjee  13th President of India
2   Smt Pratibha Devisingh Patil  12th President of India
3         DR. A.P.J. Abdul Kalam  11th President of India
4           Shri K. R. Narayanan  10th President of India
5        Dr Shankar Dayal Sharma  9th  President of India
6            Shri R Venkataraman   8th President of India
7               Giani Zail Singh   7th President of India
8      Shri Neelam Sanjiva Reddy   6th President of India
9       Dr. Fakhruddin Ali Ahmed   5th President of India
10  Shri Varahagiri Venkata Giri   4th President of India
11              Dr. Zakir Husain   3rd President of India
12  Dr. Sarvepalli Radhakrishnan   2nd President of India
13           Dr. Rajendra Prasad   1st President of India
